### ISSUE TITLE
Latency spikes

### ISSUE THREAD

**Issue by jagerman** [oxen-io/lokinet#1652]
Currently (testing with https://github.com/majestrate/loki-network/tree/introset-spacing-2021-05-19) I'm seeing fairly common latency spikes of 1s on a 1s-interval ping, like this:
```
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=837 ttl=63 time=247 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=838 ttl=63 time=246 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=839 ttl=63 time=248 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=840 ttl=63 time=248 ms
no answer yet for icmp_seq=841
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=841 ttl=63 time=1200 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=842 ttl=63 time=246 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=843 ttl=63 time=247 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=844 ttl=63 time=246 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=845 ttl=63 time=251 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=846 ttl=63 time=256 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=847 ttl=63 time=249 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=848 ttl=63 time=249 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=849 ttl=63 time=248 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=850 ttl=63 time=247 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=851 ttl=63 time=249 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=852 ttl=63 time=247 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=853 ttl=63 time=248 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=854 ttl=63 time=251 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=855 ttl=63 time=247 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=856 ttl=63 time=247 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=857 ttl=63 time=249 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=858 ttl=63 time=254 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=859 ttl=63 time=247 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=860 ttl=63 time=248 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=861 ttl=63 time=248 ms
no answer yet for icmp_seq=862
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=862 ttl=63 time=1261 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=863 ttl=63 time=247 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=864 ttl=63 time=250 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=865 ttl=63 time=247 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=866 ttl=63 time=257 ms
no answer yet for icmp_seq=867
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=867 ttl=63 time=1272 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=868 ttl=63 time=248 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=869 ttl=63 time=253 ms
no answer yet for icmp_seq=870
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=870 ttl=63 time=1258 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=871 ttl=63 time=250 ms
```

I experimented some more with different ping intervals and the latency spike seems to always be the ping interval, e.g. with a .3 interval:
```
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=43 ttl=63 time=246 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=44 ttl=63 time=254 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=45 ttl=63 time=248 ms
no answer yet for icmp_seq=46
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=46 ttl=63 time=566 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=47 ttl=63 time=386 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=48 ttl=63 time=245 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=49 ttl=63 time=251 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=50 ttl=63 time=250 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=51 ttl=63 time=282 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=52 ttl=63 time=247 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=53 ttl=63 time=248 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=54 ttl=63 time=247 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=55 ttl=63 time=252 ms
no answer yet for icmp_seq=56
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=56 ttl=63 time=559 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=57 ttl=63 time=249 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=58 ttl=63 time=261 ms
no answer yet for icmp_seq=59
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=59 ttl=63 time=569 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=60 ttl=63 time=247 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=61 ttl=63 time=246 ms
no answer yet for icmp_seq=62
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=62 ttl=63 time=575 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=63 ttl=63 time=249 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=64 ttl=63 time=248 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=65 ttl=63 time=246 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=66 ttl=63 time=248 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=67 ttl=63 time=246 ms
no answer yet for icmp_seq=68
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=68 ttl=63 time=552 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=69 ttl=63 time=250 ms
64 bytes from 172.31.0.2 (172.31.0.2): icmp_seq=70 ttl=63 time=245 ms
```

And here, again, we're getting spikes of just about 300ms, which is my ping interval.  It looks as though when the ping gets delayed it is the following ping that causes it to actually get flushed out on the network.

So perhaps we have a missing flush in some edge case?  (I know that's vague).

**Comment by jagerman** [comment]
I also tried running a 1s interval ping with a parallel .2s interval ping running and this reduced the latency spikes to .2s on the 1s ping.

I tried a .2s interval ping from a different lokinet machine and this did nothing at all on the 1s interval (i.e. it's something path specific).

**Comment by majestrate** [comment]
is this still happening?

**Comment by majestrate** [comment]
fixed in 0.9.8
